src/python/foglamp/data_purge/sqlalchemy_purge.py

```diff
@@ -0,0 +1,324 @@
+
+# Check if file exists, if not than create it
+try:
+    tmp_f = open(logs_file, 'r')
+except NameError:
+    open(logs_file, 'w').close()
+else:
+    tmp_f.close()
+
+# Table purge against
+readings_table = sqlalchemy.Table('readings', sqlalchemy.MetaData(),
+                                  sqlalchemy.Column('id', sqlalchemy.BIGINT, primary_key=True),
+                                  sqlalchemy.Column('asset_code', sqlalchemy.VARCHAR(50)),
+                                  sqlalchemy.Column('read_key', sqlalchemy.dialects.postgresql.UUID,
+                                                    default='00000000-0000-0000-0000-000000000000'),
+                                  sqlalchemy.Column('reading', sqlalchemy.dialects.postgresql.JSON, default='{}'),
+                                  sqlalchemy.Column('user_ts', sqlalchemy.TIMESTAMP(6),
+                                                    default=time.strftime('%Y-%m-%d %H:%M:%S',
+                                                                          time.localtime(time.time()))),
+                                  sqlalchemy.Column('ts', sqlalchemy.TIMESTAMP(6),
+                                                    default=time.strftime('%Y-%m-%d %H:%M:%S',
+                                                                          time.localtime(time.time()))))
+
+
+"""Methods that support the purge process. For the most part, theses methods would be replaced by either a scheduler,  
+a database API interface,  and/or proper configuration methodology. 
+"""
+
+
+def convert_timestamp(set_time: str) -> datetime.timedelta:
+    """Convert "age" in config file to timedelta. If only an integer is specified,  then 
+        the code assumes that it is already in minutes (ie age:1 means wait 1 minute) 
+    Args:
+        set_time (str): Newest amount of  time back to delete
+    Returns:
+        converted set_time to datetime.timedelta value
+    """
+    if set_time.isdigit():
+        return datetime.timedelta(minutes=int(set_time))
+    time_dict = {}
+    tmp = 0
+
+    for value in set_time.split(" "):
+        if value.isdigit() is True:
+            tmp = int(value)
+        else:
+            time_dict[value] = tmp
+
+    time_in_sec = datetime.timedelta(seconds=0)
+    time_in_min = datetime.timedelta(minutes=0)
+    time_in_hr = datetime.timedelta(hours=0)
+    time_in_day = datetime.timedelta(days=0)
+
+    for key in time_dict.keys():
+        if 'sec' in key:
+            time_in_sec = datetime.timedelta(seconds=time_dict[key])
+        elif 'min' in key:
+            time_in_min = datetime.timedelta(minutes=time_dict[key])
+        elif ('hr' in key) or ('hour' in key):
+            time_in_hr = datetime.timedelta(hours=time_dict[key])
+        elif ('day' in key) or ('dy' in key):
+            time_in_day = datetime.timedelta(days=time_dict[key])
+    return time_in_sec+time_in_min+time_in_hr+time_in_day
+
+
+def convert_sleep(set_time: str) -> int:
```

Comment:
Correct, this is not needed. The purge process will be invoked by the scheduler, run to completion and then terminate. The scheduler will rerun the purge process based on the rules defined in the schedule.